src/Simplify.elm

```diff
@@ -1310,19 +1310,18 @@ expressionBranchLocalBindings expression =
 
         Expression.LetExpression letBlock ->
             List.foldl
-                (\(Node _ letDeclaration) ->
+                (\(Node _ letDeclaration) acc ->
                     case letDeclaration of
                         Expression.LetFunction letFunctionOrValueDeclaration ->
-                            \soFar ->
-                                RangeDict.insert
-                                    (Node.range (Node.value letFunctionOrValueDeclaration.declaration).expression)
-                                    (AstHelpers.patternListBindings
-                                        (Node.value letFunctionOrValueDeclaration.declaration).arguments
-                                    )
-                                    soFar
+                            RangeDict.insert
+                                (Node.range (Node.value letFunctionOrValueDeclaration.declaration).expression)
+                                (AstHelpers.patternListBindings
+                                    (Node.value letFunctionOrValueDeclaration.declaration).arguments
+                                )
+                                acc
 
                         _ ->
-                            identity
+                            acc
```

Comment:
Just FYI: identity is curried, yes. but it's the base function so it doesn't come with performance costs

Reply:
The problem here is not `identity`, it's that the fold function is defined as taking a single argument, when it will be called with 2 in practice. It's therefore faster if we define it more explicitly with 2 arguments.